keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.netflix.spinnaker.keel.api.Verification
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.verification.VerificationContext
+import com.netflix.spinnaker.keel.api.verification.VerificationRepository
+import com.netflix.spinnaker.keel.api.verification.VerificationState
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.DELIVERY_ARTIFACT
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.DELIVERY_CONFIG
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.ENVIRONMENT
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.VERIFICATION_STATE
+import org.jooq.DSLContext
+import org.jooq.Record1
+import org.jooq.ResultQuery
+import org.jooq.Select
+import java.time.Clock
+
+class SqlVerificationRepository(
+  private val jooq: DSLContext,
+  private val clock: Clock
+) : VerificationRepository {
+
+  override fun getState(
+    context: VerificationContext,
+    verification: Verification
+  ): VerificationState? =
+    with(context) {
+      jooq
+        .select(
+          VERIFICATION_STATE.STATUS,
+          VERIFICATION_STATE.STARTED_AT,
+          VERIFICATION_STATE.ENDED_AT
+        )
+        .from(VERIFICATION_STATE)
+        .where(VERIFICATION_STATE.ENVIRONMENT_UID.eq(environmentUid))
+        .and(VERIFICATION_STATE.ARTIFACT_UID.eq(artifact.uid))
+        .and(VERIFICATION_STATE.ARTIFACT_VERSION.eq(version))
+        .fetchOneInto<VerificationState>()
```

Comment:
Fetch state for a verification in an environment. Note there's no column for the verification yet. That's next.